### ISSUE TITLE
No support for creating containers

### ISSUE THREAD

**Issue by Innectic** [faradayio/boondock#17]
The ability to create containers is super important for a library like this.

**Comment by Innectic** [comment]
Upon looking closer, there's no control of containers at all

**Comment by emk** [comment]
Yeah, `boondock` is used by [cage](https://github.com/faradayio/cage), which actually creates and controls containers using `docker-compose` (and maybe sometimes `docker`). `cage` uses `boondock` for a bunch of things that can't be done using the CLI tools.

The maintenance status of `boondock` is what Cargo would probably refer to as "passively maintained":

- If any `boondock` features used by `cage` break, Faraday's engineers should fix them quickly. We use `cage` heavily internally.
- If anything else breaks, we'll happily accept PRs, and work with people to troubleshoot and find their way around the code.
- If anybody wants new features, we're happy to accept PRs (provided there are tests and some basic docs!).
- On the off chance that anybody wants to start the process of porting this to Tokio/async, we'd be absolutely delighted. Note, however, that `boondock` uses HTTP over sockets on Unix, and it uses a platform-specific IPC method on Windows, so this is harder than it looks.

In general, the Docker HTTP interface is not especially well documented, so adding new features often involves a fair bit of time reading the code of Docker and/or `docker-compose`.

**Comment by Innectic** [comment]
Fair enough 👍 Started working on container controls last night here: https://github.com/Innectic/boondock/tree/feature/containers

**Comment by pgrinaway** [comment]
I've added some container creation logic [here](https://github.com/pgrinaway/boondock/tree/add-container-control) along with some basic tests. If people are interested I can make a PR (and add whatever you think should be added before merge).

**Comment by emk** [comment]
Oh, cool! I would definitely be happy to look at a PR.
